src/Resolvers/Microsoft.NET.Sdk.WorkloadResolver/MSBuildWorkloadSdkResolver.cs

```diff
@@ -0,0 +1,168 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+#if NET
+using Microsoft.DotNet.Cli;
+#else
+using Microsoft.DotNet.DotNetSdkResolver;
+#endif
+
+#nullable disable
+
+namespace Microsoft.NET.Sdk.WorkloadResolver
+{
+    public class MSBuildWorkloadSdkResolver : SdkResolver
+    {
+        public override string Name => "Microsoft.DotNet.MSBuildWorkloadSdkResolver";
+
+        public override int Priority => 4000;
+
+        private bool _enabled;
+
+        private IWorkloadManifestProvider _workloadManifestProvider;
+        private IWorkloadResolver _workloadResolver;
+
+
+#if NETFRAMEWORK
+        private readonly NETCoreSdkResolver _sdkResolver;
+#endif
+
+
+        public MSBuildWorkloadSdkResolver()
+        {
+            //  Put workload resolution behind a feature flag.
+            _enabled = false;
+            var envVar = Environment.GetEnvironmentVariable("MSBuildEnableWorkloadResolver");
+            if (envVar != null)
+            {
+                if (envVar.Equals("true", StringComparison.OrdinalIgnoreCase))
+                {
+                    _enabled = true;
+                }
+            }
+
+            if (!_enabled)
+            {
+                string sentinelPath = Path.Combine(Path.GetDirectoryName(typeof(MSBuildWorkloadSdkResolver).Assembly.Location), "EnableWorkloadResolver.sentinel");
```

Comment:
EnableWorkloadResolver.sentinel would be in VS hive for VS scenario. Is that ok?